lib/src/model/settings/log_preferences.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:lichess_mobile/src/model/settings/preferences_storage.dart';
+import 'package:logging/logging.dart';
+
+part 'log_preferences.freezed.dart';
+part 'log_preferences.g.dart';
+
+final logPreferencesProvider = NotifierProvider<LogPreferencesNotifier, LogPrefs>(
+  LogPreferencesNotifier.new,
+  name: 'LogPreferencesProvider',
+);
+
+class LogPreferencesNotifier extends Notifier<LogPrefs> with PreferencesStorage<LogPrefs> {
+  @override
+  @protected
+  PrefCategory get prefCategory => PrefCategory.overTheBoard;
+
+  @override
+  @protected
+  LogPrefs get defaults => LogPrefs.defaults;
+
+  @override
+  LogPrefs fromJson(Map<String, dynamic> json) => LogPrefs.fromJson(json);
+
+  @override
+  LogPrefs build() => fetch();
+
+  Future<void> setLogLevel(Level level) {
+    return save(state.copyWith(level: level));
+  }
+}
+
+const _kDefaultLevel = kDebugMode ? Level.FINE : Level.OFF;
```

Comment:
Not sure about this. But default log level could be sth else than OFF?

Reply:
Changed it to `WARNING` now